dll/cpl/timedate/internettime.c

```diff
@@ -105,6 +107,15 @@ SetNTPServer(HWND hwnd)
 
     SendDlgItemMessageW(hwnd, IDC_SERVERLIST, WM_GETTEXT, _countof(buffer), (LPARAM)buffer);
 
+    /* If the condition is true that means the user wants to update (synchronize) the time */
+    if (bBeginUpdate)
+    {
+        /* Inform the user that the synchronization is about to begin (depending on how reachable the NTP server is) */
+        StringCchPrintfW(szFormat, _countof(szFormat), SyncStatus.szSyncWait, buffer);
+        SyncStatus.lpszSyncStatus = szFormat;
```

Comment:
This isn't technically broken, but is asking for trouble.
`szFormat` will stop being valid once this function exits, therefore so will `SyncStatus.lpszSyncStatus`.
Since this member doesn't actually need to persist (it's only ever used in code exactly like this one: assign, then pass to `SetDlgItemText`), I would recommend removing it from the structure.

Reply:
Roger that. I will do this in a next PR.